### PR TITLE
[DOCS] Add concept of trained models

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/63097 and https://github.com/elastic/kibana/pull/74978

This PR provides a conceptual page about "trained models" and removes the related paragraph from the existing ["inference" page](https://www.elastic.co/guide/en/machine-learning/master/ml-inference.html).

### Preview

* https://stack-docs_1391.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-trained-models.html
* https://stack-docs_1391.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-inference.html

### CODE REVIEW

**Comment by lcawl** [comment]
@darnautov I noticed that the Kibana model management app says "Total inference trained models". Should we drop "inference" from that title to align with the API changes @benwtrent made?

**Comment by darnautov** [comment]
> @darnautov I noticed that the Kibana model management app says "Total inference trained models". Should we drop "inference" from that title to align with the API changes @benwtrent made?

@lcawl yes, makes sense, I'll create a PR 👍

File: docs/en/stack/ml/df-analytics/ml-trained-models.asciidoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+When you are satisfied with your trained model, you can use it to make
+predictions against new data. For example, you can use it in the preprocessor of
+an ingest pipeline or in a pipeline aggregation within a search query. For more
+information about this process, see <<ml-supervised-workflow>> and
+<<ml-inference>>.
+
+You can also supply trained models that are not created by {dfanalytics-job} but
+adhere to the appropriate https://github.com/elastic/ml-json-schemas[JSON schema].
+If you want to use these trained models in the {stack}, you must store them in
+{es} documents by using the {ref}/put-inference.html[create trained model API].
```
Comment (szabosteve) [inline docs/en/stack/ml/df-analytics/ml-trained-models.asciidoc:18]: In the meantime, the URLs are changed and there is a PR up for updating the titles from singular to plural.

```suggestion
{es} documents by using the {ref}/put-trained-models.html[create trained models API].
```

**Review by szabosteve — APPROVED** [review]
One minor comment, otherwise LGTM! Thanks!

File: docs/en/stack/ml/df-analytics/ml-trained-models.asciidoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[role="xpack"]
+[[ml-trained-models]]
+= Trained models
+
+experimental::[]
+
+When you use a {dfanalytics-job} to perform {classification} or {reganalysis},
+it creates a {ml} model that is trained and tested against a labelled data set.
+When you are satisfied with your trained model, you can use it to make
+predictions against new data. For example, you can use it in the preprocessor of
```
Comment (benwtrent) [inline docs/en/stack/ml/df-analytics/ml-trained-models.asciidoc:10]:
```suggestion
predictions against new data. For example, you can use it in the processor of
```
they are just called `ingest processors`

**Review by benwtrent — APPROVED** [review]

**Pull request merged**